modules/core/src/test/java/org/apache/ignite/platform/PlatformDeployServiceTask.java

```diff
@@ -671,10 +681,101 @@ public void sleep(long delayMs) {
             }
         }
 
+        /**
+         * Calculates number of registered values among the service statistics.
+         *
+         * @return Number of registered values among the service statistics. Or {@code null} if no statistics found
+         * for this service.
+         */
+        public int testNumberOfInvocations(String svcName, String histName) {
+            return ignite.compute().execute(new CountServiceMetricsTask(), new IgnitePair<>(svcName, histName));
+        }
+
         /** */
         public Object contextAttribute(String name) {
             return svcCtx.currentCallContext().attribute(name);
         }
+
+        /**
+         * Calculates number of registered values among the service statistics. Can process all service metrics or
+         * certain named one.
+         */
+        private static class CountServiceMetricsTask extends ComputeTaskSplitAdapter<IgnitePair<String>, Integer> {
+            /** {@inheritDoc} */
+            @Override public Integer reduce(List<ComputeJobResult> results) throws IgniteException {
+                long cnt = 0;
+
+                for (ComputeJobResult res : results) {
+                    if (res.isCancelled()) {
+                        throw new IgniteException("Unable to count invocations in service metrics. Job was canceled " +
+                            "on node [" + res.getNode() + "].");
+                    }
+
+                    if (res.getException() != null) {
+                        throw new IgniteException("Unable to count invocations in service metrics. Job failed on " +
+                            "node [" + res.getNode() + "]: " + res.getException().getMessage(), res.getException());
+                    }
+
+                    if (res.getData() == null)
+                        continue;
+
+                    cnt += (Long)res.getData();
```

Comment:
Obsolete cast to `Long`.

Reply:
Nope. The result has no certain type. Idea says obsolete, but not. You might try to remove the cast.

Reply:
Yes, but why `Long`? The method `reduce` returns Integer, why do we operate with long here? Should we return Long then?

Reply:
Turned to int